### PR TITLE
Added Scenario monthly profile parameter

### PR DESCRIPTION
**Pull request by jetuk**
- Added `ScenarioMonthlyProfileParameter` with two tests
- Small cython optimisation for `ArrayIndexedScenarioMonthlyFactorsParameter`

### CODE REVIEW

File: pywr/parameters/_parameters.pyx [hunk 1]
```diff
@@ -488,6 +498,34 @@ cdef class MonthlyProfileParameter(Parameter):
         return self._upper_bounds
 MonthlyProfileParameter.register()
 
+
+cdef class ScenarioMonthlyProfileParameter(Parameter):
+    def __init__(self, Scenario scenario, values, **kwargs):
```
Comment (snorfalorpagus) [inline pywr/parameters/_parameters.pyx]: Please add a docstring

File: tests/test_parameters.py [hunk 2]
```diff
@@ -128,6 +128,43 @@ def test_parameter_monthly_profile(model):
         np.testing.assert_allclose(p.value(ts, si), values[imth])
 
 
+class TestScenarioMonthlyProfileParameter:
+
+    def test_init(self, model):
+        scenario = Scenario(model, 'A', 10)
+        values = np.random.rand(10, 12)
+
+        p = ScenarioMonthlyProfileParameter(scenario, values )
```
Comment (snorfalorpagus) [inline tests/test_parameters.py]: Remove space after values.

File: tests/models/scenario_monthly_profile.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
+    "metadata": {
+        "title": "Scenario monthly profile",
+        "description": "A simple example to test ScenarioMonthlyProfileParameter",
+        "minimum_version": "0.1"
```
Comment (snorfalorpagus) [inline tests/models/scenario_monthly_profile.json]: This should be "0.2dev"

**Comment by snorfalorpagus** [comment]
This looks good. See comments above.

**Comment by jetuk** [comment]
I think I've addressed the comments.

**Pull request merged**